### DISCUSSION TITLE
How often should streamed data be sent to Deepgram

### DISCUSSION THREAD

**Discussion by alexbainbridge** [deepgram/community#463] [General help]
On https://dev.to/deepgram/getting-started-with-live-transcription-and-vuejs-1eba it suggests 

`mediaRecorder.start(250)`

On https://github.com/deepgram-devs/node-live-example/blob/main/public/client.js it suggests

`microphone.start(500);`

On https://deepgram.com/learn/how-to-add-speech-recognition-to-your-react-project it suggests

`mediaRecorder.start(1000)`

On https://developers.deepgram.com/reference/streaming it suggests

`Streaming buffer sizes should be between 20 milliseconds and 250 milliseconds of audio.`

### My tests

I was having a problem that with perhaps 1 in 10 starts (when using 150ms), Deepgram wasn't starting correctly. I haven't been able to work out the exact issue, but it seems more fault tolerant at 500ms, maybe. 

This may or may not been the cause of my problem, but either way, what IS the suggested Deepgram setting for microphone data?

Thanks!

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by davidvonthenen** [comment]
No need to delay sending the stream. You should send the stream data as soon as it's available.

**Reply by alexbainbridge** [reply]
Right, but if you do mediaRecorder.start(250), microphone will make data available every 250 milliseconds, and then send immediately to Deepgram. Perhaps I didn't phrase the question well, but its not really about how soon AFTER the data is created should we send to Deepgram, but how often should we slice the data stream (e.g. into 250 millisecond blocks) and send to Deepgram
